src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -123,19 +123,27 @@ public SnowflakeSinkServiceV2(
   @Override
   public void startTask(String tableName, String topic, int partition) {
     String partitionChannelKey = partitionChannelKey(topic, partition);
-    LOGGER.info(
-        "Opening a channel with name:{} for table name:{}, topic:{}, partition:{}",
-        partitionChannelKey,
-        tableName,
-        topic,
-        partition);
+
     // the table should be present before opening a channel so lets do a table existence check here
     createTableIfNotExists(tableName);
+
+    createOrUpdatePartitionsToChannelCache(partitionChannelKey, tableName);
+  }
+
+  /* We will always open a new channel. This will either update the TopicPartitionChannel's streamingChannel or will create a new instance of TopicPartitionChannel. */
+  private void createOrUpdatePartitionsToChannelCache(
+      final String partitionChannelKey, final String tableName) {
+    // Always open a new channel.
     SnowflakeStreamingIngestChannel partitionChannel =
         openChannelForTable(partitionChannelKey, tableName);
-    TopicPartitionChannel topicPartitionChannel =
-        new TopicPartitionChannel(partitionChannel, this.conn, tableName);
-    partitionsToChannel.putIfAbsent(partitionChannelKey, topicPartitionChannel);
+    if (partitionsToChannel.containsKey(partitionChannelKey)) {
+      // we just update the channel instance for this partition
+      partitionsToChannel.get(partitionChannelKey).updateStreamingIngestChannel(partitionChannel);
```

Comment:
Double checking we're ok with not updating offsets and the StreamingBuffer in this case where we just set a new StreamingIngestChannel

Reply:
that is correct! We will only update channel since we closed it when close() on sinktask was called. (partition rebalance)

I think it is okay to not reset buffer and processed offset. 

One of the primary reason I didnt want to recreate topicpartitonchannel object was to preserve processed offset after rebalance. so that we could start off from where we left..

Reply:
https://github.com/snowflakedb/snowflake-kafka-connector/pull/391/files#

Check topic partition channel logic. https://github.com/snowflakedb/snowflake-kafka-connector/blob/japatel-SNOW-536429-exactly-once/src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java#L117

Reply:
Now that I am thinking, there is also no harm in calling getCommittedOffsetToken in `startTask()` too to fetch the latest committed offset.

Reply:
I will handle it in subsequent PRs.

Reply:
[SNOW-542734](https://snowflakecomputing.atlassian.net/browse/SNOW-542734)